### PR TITLE
golang examples init

### PR DESCRIPTION
**Pull request by nonam3e**
## Describe the changes

This PR adds golang examples

### CODE REVIEW

File: examples/golang/ntt/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# ICICLE example: Number Theoretic Transform (NTT) in Rust
+
+## Key-Takeaway
+
+`ICICLE` provides Rust bindings to CUDA-accelerated C++ implementation of [Number Theoretic Transform](https://github.com/ingonyama-zk/ingopedia/blob/master/src/fft.md).
+
+## Usage
+
+```rust
```
Comment (LeonHibnik) [inline examples/golang/ntt/README.md:9]:
```suggestion
```go
```

File: examples/golang/msm/main.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	println(time.Since(startTime).String())
+
+	print("Generating BN254 points ... ")
+	startTime = time.Now()
+	pointsBn254Max := bn254.GenerateAffinePoints(sizeMax)
+	println(time.Since(startTime).String())
+
+	print("Generating BN254 G2 points ... ")
+	startTime = time.Now()
+	pointsBn254G2Max := bn254G2.G2GenerateProjectivePoints(sizeMax)
```
Comment (jeremyfelder) [inline examples/golang/msm/main.go:43]: This should be using Affine points

File: examples/golang/msm/main.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	println(time.Since(startTime).String())
+
+	print("Generating BLS12_377 points ... ")
+	startTime = time.Now()
+	pointsBls12377Max := bls12377.GenerateAffinePoints(sizeMax)
+	println(time.Since(startTime).String())
+
+	print("Generating BLS12_377 G2 points ... ")
+	startTime = time.Now()
+	pointsBls12377G2Max := bls12377G2.G2GenerateProjectivePoints(sizeMax)
```
Comment (jeremyfelder) [inline examples/golang/msm/main.go:58]: Here as well should be using Affine

File: examples/golang/ntt/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# ICICLE example: Number Theoretic Transform (NTT) in Rust
+
+## Key-Takeaway
+
+`ICICLE` provides Rust bindings to CUDA-accelerated C++ implementation of [Number Theoretic Transform](https://github.com/ingonyama-zk/ingopedia/blob/master/src/fft.md).
```
Comment (jeremyfelder) [inline examples/golang/ntt/README.md:5]:
```suggestion
`ICICLE` provides Golang bindings to CUDA-accelerated C++ implementation of [Number Theoretic Transform](https://github.com/ingonyama-zk/ingopedia/blob/master/src/fft.md).
```

File: examples/golang/ntt/main.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	println(time.Since(startTime).String())
+
+	print("Configuring Bls12377 NTT ... ")
+	startTime = time.Now()
+
+	streamBls12377, _ := cr.CreateStream()
+
+	cfgBls12377.Ctx.Stream = &streamBls12377
+
+	var msmResultBls12377 core.DeviceSlice
```
Comment (jeremyfelder) [inline examples/golang/ntt/main.go:87]:
```suggestion
	var nttResultBls12377 core.DeviceSlice
```

File: examples/golang/msm/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# ICICLE example: MultiScalar Multiplication (MSM) in Rust
```
Comment (jeremyfelder) [inline examples/golang/msm/README.md:1]:
```suggestion
# ICICLE example: MultiScalar Multiplication (MSM) in Golang
```

File: examples/golang/ntt/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# ICICLE example: Number Theoretic Transform (NTT) in Rust
```
Comment (jeremyfelder) [inline examples/golang/ntt/README.md:1]:
```suggestion
# ICICLE example: Number Theoretic Transform (NTT) in Golang
```

**Review by jeremyfelder — APPROVED** [review]
A few more minor changes and looks good to go

File: examples/golang/polynomials/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# ICICLE example: Polynomials in Golang
+
+`ICICLE` provides Golang bindings to CUDA-accelerated C++ implementation of [Polynomials](../../../docs/docs/icicle/polynomials/overview.md).
```
Comment (jeremyfelder) [inline examples/golang/polynomials/README.md:3]: Lets link to the deployed version

https://dev.ingonyama.com/icicle/polynomials/overview

Comment (LeonHibnik) [reply examples/golang/polynomials/README.md]: @jeremyfelder docusaurus renders the links to use a full path, so it's actually better to user relative path links when rendering locally

**Review by LeonHibnik — APPROVED** [review]

**Pull request merged**